dynamicdns.go

```diff
@@ -219,11 +225,33 @@ func (a App) checkIPAndUpdateDNS() {
 					continue
 				}
 
+				// find the record with recordId
+				recs, err := a.dnsProvider.GetRecords(a.ctx, zone)
+				if err != nil {
+					a.logger.Error("failed to get records", zap.Error(err))
+					continue
+				}
+
+				var record libdns.Record
+				for _, rec := range recs {
+					if rec.Name == domain && rec.Type == recordType(ip) {
+						record = rec
+						break
+					}
+				}
```

Comment:
If it doesn't find one, shouldn't it use `recordType(ip)` as the Type and `domain` as the Name (i.e. set defaults for `record`)?